### PR TITLE
Fix/migrate3-deprecate

### PR DESCRIPTION
**Pull request by johnpapa**
* Old versions of Peacock did not store the peacock color or stored it in a memento.
   * Version 3+ of Peacock stores the color in the settings.
   * If `peacock.color` does not exist and we find either memento or a derived color (activity bar, title bar, status bar),
   * remove the memento, and apply the color to the settings.
   * @deprecated since version 3.0.
   * Will be deleted in version 4.0 and once v3.0 users have migrated

As peacock is installed 1.4 million instances of vscode, I did not want to have a breaking change to existing users. This logic was put in place to help automatically migrate users from older versions to newer ones. I will remove this when we go to version 4, which may be about time.

## Why does it exist?

1. Without this logic, users of older versions of peacock wouldnt work. 
2. If a user with an older version of peacock updates Peacock, this logic will auto migrate them. Which means they wont need this logic moving forward. 

The idea here was to help users migrate, without them needing to do anything. Thus, after time, most users would already be migrated. This plan seems to have worked very well as this is the first issue in more than a year on this topic.

## Is it needed anymore?

I could remove this logic and shift to version 4 now, as it would be a breaking change. And my gut says it is time (almost 2 years since v3).  This is the only change driving a new version at this time, so that Is something to consider.

I could also modify the logic to not check for the existing derived colors, which is what you are running into. This logic exists as a reasonable way for Peacock to determine if it should apply a color. But w/o this logic, it would be difficult to make a reasonable color choice.

tldr; I am deprecating the code.

### CODE REVIEW

**Comment by johnpapa** [comment]
closes #480
